src/promptflow-devkit/promptflow/_sdk/operations/_flow_operations.py

```diff
@@ -1121,7 +1119,46 @@ def infer_signature(self, entry: Callable) -> dict:
         :rtype: dict
         """
         # TODO: should we support string entry? If so, we should also add a parameter to specify the working directory
-        flow_meta, _, _ = self._infer_signature(entry=entry)
+        if isinstance(entry, Prompty):
+            from promptflow._sdk.schemas._flow import ALLOWED_TYPES
+            from promptflow.contracts.tool import ValueType
+            from promptflow.core._model_configuration import PromptyModelConfiguration
+
+            flow_meta = {
+                "inputs": entry._data.get("inputs", {}),
+                "outputs": entry._data.get("outputs", {"output": "string"}),
+            }
+            init_dict = {}
+            for field in fields(PromptyModelConfiguration):
+                filed_type = type(field.type).__name__
+                init_dict[field.name] = {"type": filed_type if filed_type in ALLOWED_TYPES else ValueType.OBJECT.value}
+                if field.default != MISSING:
+                    init_dict[field.name]["default"] = field.default
+            flow_meta["init"] = init_dict
+            format_signature_type(flow_meta)
+        elif isinstance(entry, Flow):
+            # Get dag flow infer signature
+            executable_flow = entry._init_executable()
```

Comment:
please avoid init executable in sdk if possible. Such usages usually indicates a language related logic which should be put in proxies

Reply:
remove support for dag flow